broker-core/src/main/java/org/apache/qpid/server/transport/NonBlockingConnection.java

```diff
@@ -691,4 +693,12 @@ public String getSelectedHost()
     {
         return _selectedHost;
     }
+
+    public boolean detectConnectionLooping() { return _port.getContextValue(Boolean.class, AmqpPort.PORT_DETECT_CONNECTION_LOOPING); }
```

Comment:
why method is public?
IMHO, package protected visibility is better for the method.

As it is only used in `NonBlockingConnectionTLSDelegate`, it make sense to move it there...
The evaluation of the context variables can be added into `NonBlockingConnectionTLSDelegate` constructor and stored in field, as context variable evaluation  is quite expensive operation. It impacts the performance...

The name of method\variable can be changed to `enableDiagnosisOfSslEngineLooping`...